src/Hedgehog/Random.fs

```diff
@@ -3,6 +3,7 @@
 open Hedgehog.Numeric
 
 /// A generator for random values of type 'a
+[<Struct>]
```

Comment:
@Porges, I'm taking this one out (along with some tiny nitpicks I'm doing on the branch before merging). It might be nice to have `Random<'a>` marked as struct, but then it changes its semantics; it'll become pass-by-value, allocated on thread stack instead of the managed heap, etc. (breaking change, more or less).

It would be absolutely fine doing this separately though, and do a major version bump for it. :100:

Reply:
Fine with me 🙂

Reply:
On the other hand, it *will* be a major version bump because of `LazyList` -> `seq` so perhaps now is the right moment to go for it.